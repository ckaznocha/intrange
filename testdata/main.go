package main

func main() {
	for i := 2; i < 10; i++ {
	}

	for i := 0; i < 10; i += 2 {
	}

	for i := 0; i < 10; i++ {
		i += 1
	}

	for i := 0; i < 10; i++ {
		i++
	}

	for i := 0; i < 10; i++ {
		i = i + 1
	}

	for i := 0; i < 10; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := uint32(0); i < 10; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0x0; i < 10; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < 10; i += 1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < 10; i += 0x1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < 10; i = i + 1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < 10; i = i + 0x1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < 10; i = 1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < 10; i = 0x1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; 10 > i; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0x0; 10 > i; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; 10 > i; i += 1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; 10 > i; i += 0x1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; 10 > i; i = i + 1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; 10 > i; i = i + 0x1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; 10 > i; i = 1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; 10 > i; i = 0x1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	const x = 10

	for i := 2; i < x; i++ {
	}

	for i := 0; i < x; i += 2 {
	}

	for i := 0; i < x; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := uint32(0); i < uint32(x); i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0x0; i < x; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < x; i += 1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < x; i += 0x1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < x; i = i + 1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < x; i = i + 0x1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < x; i = 1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < x; i = 0x1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; x > i; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; x > i; i += 1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; x > i; i += 0x1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; x > i; i = i + 1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; x > i; i = i + 0x1 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; x > i; i = 1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; x > i; i = 0x1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}
}
