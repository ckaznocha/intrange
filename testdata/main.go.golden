package main

import (
	"testing"

	"google.golang.org/protobuf/reflect/protoreflect"
)

func calculate(i int) int32 {
	return int32(i)
}

func main() {
	for i := 2; i < 10; i++ {
	}

	for i := 0; i < 10; i += 2 {
	}

	for i := 0; i < 10; i++ {
		i += 1
	}

	for i := 0; i < 10; i++ {
		i++
	}

	for i := 0; i < 10; i++ {
		i = i + 1
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range calculate(10) { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range uint32(10) { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 10 { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	const x = 10

	for i := 2; i < x; i++ {
	}

	for i := 0; i < x; i += 2 {
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range 1*x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range calculate(1*x) { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range uint32(x) { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range x { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	var b *testing.B
	for i := range b.N { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := range b.N { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	var n int
	for i := 0; i < n; i++ {
		n--
	}

	for i := 0; i < n; i++ {
		n++
	}

	// Example from https://github.com/ckaznocha/intrange/issues/12
	var what string
	for i := 0; i < len(what); i++ {
		if what[i] == 'v' && i+1 < len(what) && what[i+1] >= '0' && what[i+1] <= '9' {
			what = what[:i] + what[i+1:]
		}
	}

	for i := range len(what) { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	var t struct{ n int }
	for i := range t.n { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < t.n; i++ {
		t.n++
	}

	var s []int
	for i := range len(s) { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < len(s); i++ {
		s = append(s, 4)
	}

	var m map[int]int
	for i := range len(m) { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < len(m); i++ {
		m[4] = 4
	}

	var t2 struct{ m map[int]int }
	for i := range len(t2.m) { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
	}

	for i := 0; i < len(t2.m); i++ {
		t2.m[4] = 4
	}
}

// https://github.com/ckaznocha/intrange/issues/16
func issue16(service protoreflect.ServiceDescriptor) {
	for i := range service.Methods().Len() { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
		print(i)
	}
}

func opReEval_IndexExpressions_ArrayLike(n []int) {
	for i := range n[0] { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
		print(i)
	}

	for i := 0; i < n[0]; i++ {
		n[0]++
	}

	for i := 0; i < n[0]; i++ {
		n[0] = 5
	}
}

func opReEval_IndexExpressions_Map(n map[string]int) {
	for i := range n["N"] { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
		print(i)
	}

	for i := 0; i < n["N"]; i++ {
		n["N"]++
	}

	for i := 0; i < n["N"]; i++ {
		n["N"] = 5
	}
}

func issue27() {
	var someSlice []interface{}

	for i := range someSlice { // want "for loop can be changed to `i := range someSlice`"
		print(i)
	}

	for n := range someSlice { // want "for loop can be changed to `n := range someSlice`"
		print(n)
	}

	for range someSlice { // want "for loop can be changed to `range someSlice`"
	}

	for range someSlice { // want "for loop can be changed to `range someSlice`"
	}

	for i := range notLen(someSlice) {
		print(i)
	}

	for i := range len(someSlice) / 2 {
		print(i)
	}

	for i := range someSlice {
		print(i)
	}

	for i, _ := range someSlice {
		print(i)
	}

	var someArray [2]interface{}

	for i := range someArray { // want "for loop can be changed to `i := range someArray`"
		print(i)
	}

	for range someArray { // want "for loop can be changed to `range someArray`"
	}

	for range someArray { // want "for loop can be changed to `range someArray`"
	}

	for i := range notLen(someArray) {
		print(i)
	}

	for i := range len(someArray) / 2 {
		print(i)
	}

	for i := range someArray {
		print(i)
	}

	for i, _ := range someArray {
		print(i)
	}

	var someMap map[int]interface{}
	for i := range len(someMap) {
		print(i)
	}

	for _ = range len(someMap) {
	}

	for range len(someMap) {
	}

	for i := range notLen(someMap) {
		print(i)
	}

	for i := range len(someMap) / 2 {
		print(i)
	}

	for i := range someMap {
		print(i)
	}

	for i, _ := range someMap {
		print(i)
	}
}

func notLen(any) int {
	return 0
}

func issue33() {
	var i int
	for i = 0; i < 10; i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)\nBecause the key is not part of the loop's scope, take care to consider side effects.`
	}
}
